### DISCUSSION TITLE
AzerothCore Modules Announcements

### DISCUSSION THREAD

**Discussion by virgo77** [azerothcore/azerothcore-wotlk#11134] [General]
Hi there!

I'm freshly using this module: [mod-skip-dk-starting-area](https://github.com/azerothcore/mod-skip-dk-starting-area/)

And I noticed that each time I connect to the server, I can see a yellow line telling that the AzerothCore is running this mod.

As I use more mods now, I would like some help on how I can modify the .cpp file in the src folder of the other mods, so the core would be able to announce the modules I installed on it.

In the DK Skip Area mod, I found this file: https://github.com/azerothcore/mod-skip-dk-starting-area/blob/master/src/SkipDK.cpp

And in this file, I found this:
`{
public:
    AzerothCore_skip_deathknight_announce() : PlayerScript("AzerothCore_skip_deathknight_announce") { }

    void OnLogin(Player* Player)
    {
        if (sConfigMgr->GetOption<bool>("Skip.Deathknight.Starter.Announce.enable", true) && (sConfigMgr->GetOption<bool>("Skip.Deathknight.Starter.Enable", true) || sConfigMgr->GetOption<bool>("Skip.Deathknight.Optional.Enable", true)))
        {
            ChatHandler(Player->GetSession()).SendSysMessage("This server is running the |cff4CFF00Azerothcore Skip Deathknight Starter |rmodule.");
        }
    }
};`

And as I am not comfy with C++, I would like some help and some explainations on about the informations in this section, like:
-  _AzerothCore_skip_deathknight_announce() : PlayerScript("AzerothCore_skip_deathknight_announce") { }_
   and how to modify this text "skip_deathknight_announce", because I didn't find any variable about this...
- _if (sConfigMgr->GetOption<bool>("Skip.Deathknight.Starter.Announce.enable", true) && (sConfigMgr->GetOption<bool>("Skip.Deathknight.Starter.Enable", true) || sConfigMgr->GetOption<bool>("Skip.Deathknight.Optional.Enable", true)))_
    I'm sure that this function is here for something special

I would like to try to add this "public" section inside my other mods "src\modfile.cpp" to make the core announce these mods. But I don't want to break it, and I don't want to make some mistakes. Maybe I can just copy this little section and make simple changes? Any suggestions, advices?

If some people here can explain me how to do this, I would be glad :)

Thanks a lot

C you soon

V.

**Comment by Kitzunu** [comment]
check the config, and turn off Skip.Deathknight.Starter.Announce.enable

**Reply by virgo77** [reply]
erm, it is not what I want. I want to enable the announce of the mods I use, as it is not configured on each mods.
So I would like to add the announce code on each mod, so ACore would be able to show each mod announce at launch:
ah-bot
transmog

thanks for help :)

V.
